gridcapa-cse-valid-app/src/main/java/com/farao_community/farao/cse_valid/app/dichotomy/DichotomyRunner.java

```diff
@@ -6,86 +6,83 @@
-import com.farao_community.farao.dichotomy.shift.SplittingFactors;
 import com.farao_community.farao.rao_runner.api.resource.RaoResponse;
 import com.farao_community.farao.rao_runner.starter.RaoRunnerClient;
-import com.powsybl.glsk.api.GlskDocument;
-import com.powsybl.iidm.network.Country;
 import com.powsybl.iidm.network.Network;
-
 import org.slf4j.Logger;
 import org.springframework.stereotype.Service;
 
-import java.util.Map;
-import java.util.TreeMap;
-
 /**
  * @author Theo Pascoli {@literal <theo.pascoli at rte-france.com>}
  * @author Ameni Walha {@literal <ameni.walha at rte-france.com>}
+ * @author Oualid Aloui {@literal <oualid.aloui at rte-france.com>}
  */
 @Service
 public class DichotomyRunner {
     private static final RangeDivisionIndexStrategy INDEX_STRATEGY_CONFIGURATION = new RangeDivisionIndexStrategy(false);
     private static final String DICHOTOMY_PARAMETERS_MSG = "Minimum dichotomy index: {}, Maximum dichotomy index: {}, Dichotomy precision: {}";
     private static final double DEFAULT_DICHOTOMY_PRECISION = 50;
     private static final int DEFAULT_MIN_INDEX = 0;
-    private static final double SHIFT_TOLERANCE = 1;
 
     private final FileImporter fileImporter;
     private final FileExporter fileExporter;
     private final RaoRunnerClient raoRunnerClient;
     private final Logger businessLogger;
+    private final CseValidNetworkShifter cseValidNetworkShifter;
 
-    public DichotomyRunner(FileImporter fileImporter, FileExporter fileExporter, RaoRunnerClient raoRunnerClient, Logger businessLogger) {
+    public DichotomyRunner(FileImporter fileImporter,
+                           FileExporter fileExporter,
+                           RaoRunnerClient raoRunnerClient,
+                           Logger businessLogger,
+                           CseValidNetworkShifter cseValidNetworkShifter) {
         this.fileImporter = fileImporter;
         this.fileExporter = fileExporter;
         this.raoRunnerClient = raoRunnerClient;
         this.businessLogger = businessLogger;
+        this.cseValidNetworkShifter = cseValidNetworkShifter;
     }
 
-    public DichotomyResult<RaoResponse> runImportCornerDichotomy(CseValidRequest cseValidRequest, TTimestamp timestamp) {
-        int npAugmented = timestamp.getMNII().getV().intValue();
-        int np = timestamp.getMiBNII().getV().intValue() - timestamp.getANTCFinal().getV().intValue();
+    public DichotomyResult<RaoResponse> runImportCornerDichotomy(TTimestampWrapper timestampWrapper, CseValidRequest cseValidRequest) {
+        int npAugmented = timestampWrapper.getMniiIntValue();
+        int np = timestampWrapper.getMibniiIntValue() - timestampWrapper.getAntcfinalIntValue();
         double maxValue = (double) npAugmented - np;
-        Network network = importNetworkFile(cseValidRequest);
+        Network network = fileImporter.importNetwork(cseValidRequest.getCgm().getFilename(), cseValidRequest.getCgm().getUrl());
```

Comment:
Use only the URL directly or add a new import method that uses CseValidFileResource to make it more readable